CMakeLists.txt

```diff
@@ -274,30 +278,61 @@ IF(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     SET_TARGET_PROPERTIES(${PROJECT_TARGET_BIN} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_SOURCE_DIR}/resources/Info.plist")
 ENDIF()
 
+# Fixes for Visual Studio 2017 immediate build\execution\debug directly opening CMake folder without project
+# Initialize IS_VS_2017CMAKE to 0, it will be override by Visual Studio cmake with -D command line parameter
+SET(IS_VS_2017CMAKE 0 CACHE BOOL "Executed from VS2017 IDE")
+IF(IS_VS_2017CMAKE EQUAL 1)
+    SET(CONFIGURATION_NAME "${TARGET_ARCH}-${CMAKE_BUILD_TYPE}")
+    SET(EXECUTABLE_OUTPUT_PATH "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}/bin")
+    SET(LIBRARY_OUTPUT_PATH "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}/bin/lib")
+    SET(INSTALL_BIN_DIR "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}/bin")
+    SET(INSTALL_DOC_DIR "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}")
+    SET(INSTALL_RES_DIR "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}/res")
+    SET(INSTALL_MO_DIR "${CMAKE_SOURCE_DIR}/bin/${CONFIGURATION_NAME}/po/en")
+    ADD_CUSTOM_COMMAND(
+        TARGET ${PROJECT_TARGET_BIN} POST_BUILD
+        COMMAND cmake -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_install.cmake"
+        WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+        COMMENT "Copying files to output directory" VERBATIM
+    )
+ENDIF()
 # Finally set ups for multiplatform INSTALL
 # Binary
 IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     INSTALL(PROGRAMS
-            "${CMAKE_SOURCE_DIR}/bin/${PROJECT_TARGET_BIN}"
+            "${EXECUTABLE_OUTPUT_PATH}/${PROJECT_TARGET_BIN}"
             DESTINATION "${INSTALL_BIN_DIR}")
 ELSEIF(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     INSTALL(TARGETS
             "${PROJECT_TARGET_BIN}"
             BUNDLE DESTINATION .)
 ELSEIF(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-    INSTALL(PROGRAMS
-            "${CMAKE_SOURCE_DIR}/bin/${PROJECT_TARGET_BIN}.exe"
-            DESTINATION "${INSTALL_BIN_DIR}")
-
+    # For VS2017 with cmake mmex.exe it's already in the right place
```

Comment:
If you refer to other VS versions, it's rounded inside a  "IF(NOT IS_VS_2017CMAKE EQUAL 1)" below, so for CLI still works fine